.github/workflows/nm-build.yml

```diff
@@ -118,15 +124,35 @@ jobs:
               with:
                 python: ${{ inputs.python }}
                 venv: ${{ env.VENV_BASE }}
-                pypi: ${{ secrets.NM_PRIVATE_PYPI_LOCATION }}
+
+            # GCP
+            - name: 'Authenticate to Google Cloud'
+              id: auth
+              uses: google-github-actions/auth@v2.1.3
+              with:
+                  project_id: ${{ secrets.GCP_PROJECT }}
+                  workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+                  service_account: ${{ secrets.NM_PYPI_SA }}
+
+            - name: 'Set up Cloud SDK'
+              uses: 'google-github-actions/setup-gcloud@v2'
+              with:
+                  version: '>= 473.0.0'
+
+            - name: copy whl and source distribution
```

Comment:
Should we not copy the whls and tar.gz to gcp here? We're only copying to gcp if all tests pass and push_to_pypi is set to True by calling .github/workflows/nm-upload-assets-to-gcp.yml right?

BTW I think we need a similar action as nm-upload-assets-to-gcp.yml to upload the whls/tar.gz to public pypi.

Reply:
yeah, let's add uploading to "public pypi" as a separate PR.